src/DataValues/DecimalValue.php

```diff
@@ -360,13 +350,20 @@ public function getArrayValue() {
 	}
 
 	/**
-	 * Constructs a new instance of the DataValue from the provided data.
-	 * This can round-trip with @see getArrayValue
+	 * Constructs a new instance from the provided data. Required for @see DataValueDeserializer.
+	 * This is expected to round-trip with @see getArrayValue.
+	 *
+	 * @deprecated since 0.8.3. Static DataValue::newFromArray constructors like this are
+	 *  underspecified (not in the DataValue interface), and misleadingly named (should be named
+	 *  newFromArrayValue). Instead, use DataValue builder callbacks in @see DataValueDeserializer.
```

Comment:
> Wrong.

It appears you did not read what I wrote, or just don't want to understand what I meant. I find this really frustrating. Again: since 2013 the detailed specification of how all newFromArray methods must behave was not in the DataValue interface but in DataValueDeserializer. How do you call this, if not a circular dependency?